SoftLayer/managers/dedicated_host.py

```diff
@@ -317,18 +318,57 @@ def _get_backend_router(self, locations, item):
                 if category['categoryCode'] == 'dedicated_host_disk':
                     disk_capacity = capacity['capacity']
 
+        for hardwareComponent in item['bundleItems']:
+            if hardwareComponent['keyName'].find("GPU") != -1:
```

Comment:
instead of creating 4 variables, it would makes sense to do something like this:

```python
gpuComponent =  {
                                    'hardwareComponentModel': {
                                        'hardwareGenericComponentModel': {
                                            'id': hardwareGenericComponentModelId,
                                            'hardwareComponentType': {
                                                'keyName': hardwareComponentTypeKeyName
                                            }
                                        }
                                    }
                                },
                                {
                                    'hardwareComponentModel': {
                                        'hardwareGenericComponentModel': {
                                            'id': hardwareGenericComponentModelId,
                                            'hardwareComponentType': {
                                                'keyName': hardwareComponentTypeKeyName
                                            }
                                        }
                                    }
                                }

```

Then just use `gpuComponent` further down.

Reply:
Done